keps/sig-node/5067-pod-generation/README.md

```diff
@@ -0,0 +1,1072 @@
+yet have a `metadata.generation` will now have a `metadata.generation` of 1.
+
+If a pod that has a `metadata.generation` set or incremented via the new API server is
+later updated by an older API server, the older API server will not modify the
+`metadata.generation` field and it will stay fixed at its current value. That means that 
+if there is version skew between multiple apiservers, the `metadata.generation` may or may
+not be incremented. To address this, we will not feature-gate the logic in the
+apiserver that increments `metadata.generation`. That means that by the time 
+ObservedGeneration goes to beta, there will be 2 versions of apiservers updating
+the `metadata.generation` field, removing the issue.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml` . This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[ `prod-readiness-approvers` ](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+* [x] Feature gate (also fill in values in `kep.yaml`)
+  + Feature gate name: PodObservedGenerationTracking
+  + Components depending on the feature gate: Kubelet, Node Lifecycle Controller, Scheduler
+
+###### Does enabling the feature change any default behavior?
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+The pod's `metadata.generation` field is currently unset by default and
+`status.observedGeneration` will be a new field in the pod status, so the feature
```

Comment:
so if feature was disabled when pod was created, and then enabled, will we start incrementing from `0`?

Reply:
yeah, the first update to the already-existing pod after the feature is enabled should set `generation` to 1